qadence_libs/qinfo_tools/qng.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+from typing import Callable
+
+import torch
+from qadence import QuantumCircuit
+from torch.optim.optimizer import Optimizer, required
+
+from qadence_libs.qinfo_tools.qfi import get_quantum_fisher, get_quantum_fisher_spsa
+
+
+class QNG(Optimizer):
+    """Implements the Quantum Natural Gradient Algorithm."""
+
+    def __init__(
+        self,
+        params: tuple | torch.Tensor,
+        circuit: QuantumCircuit = required,
+        lr: float = required,
+        beta: float = 10e-3,
+    ):
+        defaults = dict(circuit=circuit, lr=lr, beta=beta)
+        super(QNG, self).__init__(params, defaults)
+
+    def __setstate__(self, state):  # type: ignore
+        super().__setstate__(state)
+
+    def step(self, closure: Callable | None = None) -> torch.Tensor:
```

Comment:
```suggestion
    def step(self, closure: callable | None = None) -> torch.Tensor:
```

Reply:
Is this right? The `Callable` type from `typing.Callable` is with capital C (https://docs.python.org/3/library/typing.html#annotating-callable-objects)

Reply:
True enough but since you do:

```python
from __future__ import annotations
```
I thought there was a `callable` type. Maybe I'm wrong here.

Reply:
```suggestion
    def step(self, closure: Callable | None = None) -> Tensor:
```